### PR TITLE
`railway link` improvements

### LINKED ISSUE #550: [Feature Request] Specify project name inside railway link
`railway link` only supports the projectID. `railway run` supports ID or name. Railway link should support projectID or name interchangibly

### PR DESCRIPTION
**Pull request by Milo123459**
Fixes #550 

This allows you to specify a project name now. It also handles many more edge cases.

image

### CODE REVIEW

**Comment by Milo123459** [comment]
Line 220 has nothing to do with `get_project`...

File: src/commands/link.rs [hunk 1]
```diff
@@ -23,59 +22,137 @@ pub struct Args {
...
+                team.node
+                    .projects
+                    .edges
+                    .iter()
+                    .any(|proj| proj.node.id.to_lowercase() == project.to_lowercase())
+            }) {
+                fake_select("Select a team", &team.node.name);
+                Team::Team(&team.node)
+            } else {
+                dbg!("ouch");
```
Comment (coffee-cup) [inline src/commands/link.rs:73]: 🔫

**Review by coffee-cup — APPROVED** [review]
The link command function is getting fairly large and unwieldy. I'll leave this up to you, but we might want to split it up so it reads like

let team = select_team(...)
let project = select_project(...)
let environment = select_environment(...)
let service = select_service(...)

With all the logic for selecting abstracted away.

File: src/commands/link.rs [hunk 2]
```diff
@@ -23,59 +22,137 @@ pub struct Args {
...
+    .await?
+    .me;
 
-    let project = NormalisedProject::from(if let Some(project_id) = args.project_id {
-        let fetched_project = get_project(&client, &configs, project_id).await?;
-        // fake_select is used to mimic the user providing input in the terminal
-        // just for detail
-        if let Some(team) = fetched_project.clone().team {
-            fake_select("Select a team", &team.name);
+    let regex =
```
[the remark this answers is not in the document]
Comment (Milo123459) [reply src/commands/link.rs:49]: It can't be a static or a const so it has to be defined in the function

**Pull request merged**